src/qt/rpcconsole.cpp

```diff
@@ -468,6 +471,73 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
     }
 }
 
+/**
+ * @brief Executes the console-only command "generate".
+ * @param parsed_command  A vector of strings with command and parameters, usually generated by RPCExecutor::parseHelper
+ * @param wallet_model  WalletModel to use for the command
+ * @return True if the command was executed, false otherwise.
+ */
+bool RPCExecutor::executeConsoleGenerate(const std::vector<std::string>& parsed_command, const WalletModel* wallet_model, const bool exec_help)
+{
+    // Initialize default parameters if missing
+    const std::string nblocks{parsed_command.size() > 1 ? parsed_command[1] : "1"};
+    const std::string maxtries{parsed_command.size() > 2 ? parsed_command[2] : "1000000"};
+
+    // Handle some special cases...
+    // Default to console help generate if more than 3 parameters or if "help generate" was called
+    if (parsed_command.size() > 3 || exec_help) {
+        executeConsoleHelpGenerate();
+        return true;
+    }
+    // Fail if we are on mainnet, to avoid generating addresses for blocks that will not be generated
+    if (Params().GetChainType() == ChainType::MAIN) {
```

Comment:
Better to fail on all networks expect regtest. It might be possible to mine on a custom signet too, but anyone who knows how to make that can use the RPC.

Reply:
Agree, I'll change it soon